src/synthetix/StakingRewards.t.sol

```diff
@@ -270,16 +270,213 @@ contract StakingRewardsTest is Test {
+
+        skip(5 days);
+
+        rewards.exit();
+
+        uint256 rewardLeft = totalReward % 7 days + 6 days * (totalReward / 7 days) % 5 days; // dust lost due to rewardRate being rounded down
+        uint256 rewardPaid = totalReward - rewardLeft;
+        assertEq(rewardGem.balanceOf(address(this)), rewardPaid);
+        assertEq(rewardGem.balanceOf(address(rewards)), rewardLeft);
+        assertEq(gem.balanceOf(address(this)), totalStake);
+        assertEq(gem.balanceOf(address(rewards)), 0);
+    }
+
+    function testSetSameRewardsDurationOnActiveDistribution() public {
+        uint256 totalStake = 100 * WAD;
+        uint256 totalReward = 123 * WAD;
+        setupStakingToken(100 * WAD);
+        rewards.stake(100 * WAD);
+        setupReward(totalReward);
+        assertEq(rewardGem.balanceOf(address(this)), 0);
+        assertEq(rewardGem.balanceOf(address(rewards)), totalReward);
+        assertEq(gem.balanceOf(address(this)), 0);
+        assertEq(gem.balanceOf(address(rewards)), totalStake);
+        assertEq(rewards.rewardPerTokenStored(), 0);
+        assertEq(rewards.lastUpdateTime(), block.timestamp);
+        assertEq(rewards.rewardRate(), totalReward / 7 days);
+        assertEq(rewards.periodFinish(), block.timestamp + 7 days);
+        assertEq(rewards.rewardsDuration(), 7 days);
+
+        rewards.setRewardsDuration(7 days); // same duration as before
+
+        assertEq(rewards.rewardPerTokenStored(), 0);
+        assertEq(rewards.lastUpdateTime(), block.timestamp);
+        assertEq(rewards.rewardRate(), totalReward / 7 days);
+        assertEq(rewards.periodFinish(), block.timestamp + 7 days);
+        assertEq(rewards.rewardsDuration(), 7 days);
+
+        rewards.getReward();
+
+        assertEq(rewardGem.balanceOf(address(this)), 0);
+        assertEq(rewardGem.balanceOf(address(rewards)), totalReward);
+
+        skip(1 days);
+
+        rewards.setRewardsDuration(7 days); // same duration as before
+
+        assertEq(rewards.rewardPerTokenStored(), 1 days * (totalReward / 7 days) * WAD / totalStake);
+        assertEq(rewards.lastUpdateTime(), block.timestamp);
+        assertEq(rewards.rewardRate(), 6 days * (totalReward / 7 days) / 7 days);
+        assertEq(rewards.periodFinish(), block.timestamp + 7 days);
+        assertEq(rewards.rewardsDuration(), 7 days);
+
+        rewards.exit();
+
+        uint256 rewardDust = totalReward % 7 days; // dust lost due to rewardRate being rounded down
+        uint256 rewardLeft = 6 days * (totalReward / 7 days) + rewardDust;
+        uint256 rewardPaid = totalReward - rewardLeft;
+        assertEq(rewardGem.balanceOf(address(this)), rewardPaid);
+        assertEq(rewardGem.balanceOf(address(rewards)), rewardLeft);
+        assertEq(gem.balanceOf(address(this)), totalStake);
+        assertEq(gem.balanceOf(address(rewards)), 0);
+    }
+
+    function testSetRewardsDurationAndNotifyRewardAmountCallOrderOnActiveDistribution() public {
```

Comment:
I haven't gone deep on this test yet but I'd think you could use the vm.snapshot functionality and rollback to the prev state so avoiding to recreate the farm.

Reply:
Good point, done.